docs/community/maintainers-guide/releasing.rst

```diff
@@ -0,0 +1,224 @@
+..
+    This file is part of Invenio.
+    Copyright (C) 2020 CERN.
+
+    Invenio is free software; you can redistribute it and/or modify it
+    under the terms of the MIT License; see LICENSE file for more details.
+
+Releasing
+=========
+
+In general, you have two types of releases:
+
+- **Major/Minor-level release** - Often these releases add significant new
+  features and needs to go through extensive testing.
+- **Patch-level release** - Often these releases fix security issues or
+  important bugs.
+
+For major/minor-level releases, you'll often be working only with the
+``master``-branch and only make a single PyPI release. For patch-level
+releases you'll often be working with multiple maintenance branches and make
+several PyPI releases for all currently maintained versions of Invenio.
```

Comment:
Minor: might be worth mention that this is what's called *a backport release*?

Reply:
It's not really a backport, since a backport is a new feature added in a new release, that is being ported to work on an older release. We don't do that. We make bug fixes on all supported releases.